docs/USAGE-GUIDE.md

```diff
@@ -654,6 +654,18 @@ If the read end of the pipe is closed unexpectedly, writing to the pipe will rai
 a SIGPIPE signal. **s2n-tls does NOT handle SIGPIPE.** A SIGPIPE signal will cause
 the process to terminate unless it is handled or ignored by the application.
 
+**WARNING (TLS v1.2 ONLY):** Enabling Nagle's algorithm can increase handshake latency
+* Enabling Nagle's algorithm can increase the handshake latency. To avoid this behaviour, please disable Nagle's at the socket level; this can be done with the snippet displayed below. Please note this is only applicable to TLS v1.2.
```

Comment:
I would include a brief description of Nagle's algorithm here and mention that it is on by default.

Reply:
I'll just link to rfc896